hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/fair/QueueManager.java

```diff
@@ -272,6 +272,16 @@ private FSQueue createNewQueues(FSQueueType queueType,
       FSParentQueue newParent = null;
       String queueName = i.next();
 
+      // Check if child policy is allowed
```

Comment:
Should this check be in setPolicy or the FSQueue constructor instead? 

For instance, FSLeafQueue#setPolicy already checks if the level is appropriate. This brings up another point - do we need this check of parent-child policies AND the depth? Should we get rid of depth either in this JIRA or a follow-up?